File: linkdisk_flutter/src/LinkDisk/network/TransferTask.java
// TransferTask.java
package LinkDisk.network;

import java.io.Serializable;
import java.util.UUID;

public class TransferTask implements Serializable {
    private String taskId;
    private String targetIp;
    private String fileName;
    private long totalSize;
    private long transferred;
    private String status; // waiting, running, paused, completed, failed, cancelled
    private String type; // upload, download
    private long createTime;
    
    public TransferTask() {
        this.taskId = UUID.randomUUID().toString();
        this.createTime = System.currentTimeMillis();
        this.transferred = 0;
    }
    
    public TransferTask(String targetIp, String fileName, long totalSize, String type) {
        this();
        this.targetIp = targetIp;
        this.fileName = fileName;
        this.totalSize = totalSize;
        this.type = type;
    }
    
    // Getters and Setters
    public String getTaskId() { return taskId; }
    public void setTaskId(String taskId) { this.taskId = taskId; }
    
    public String getTargetIp() { return targetIp; }
    public void setTargetIp(String targetIp) { this.targetIp = targetIp; }
    
    public String getFileName() { return fileName; }
    public void setFileName(String fileName) { this.fileName = fileName; }
    
    public long getTotalSize() { return totalSize; }
    public void setTotalSize(long totalSize) { this.totalSize = totalSize; }
    
    public long getTransferred() { return transferred; }
    public void setTransferred(long transferred) { this.transferred = transferred; }
    
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    
    public long getCreateTime() { return createTime; }
    
    public int getProgress() {
        if (totalSize == 0) return 0;
        return (int) ((transferred * 100) / totalSize);
    }
}

File: linkdisk_flutter/副本lib/network/TransferTask.dart
import 'dart:uuid';
import 'package:intl/intl.dart';

class TransferTask {
  late String taskId;
  String? targetIp;
  String? fileName;
  int totalSize = 0;
  int transferred = 0;
  String? status; // waiting, running, paused, completed, failed, cancelled
  String? type; // upload, download
  late int createTime;

  // 无参构造
  TransferTask() {
    taskId = Uuid().v4();
    createTime = DateTime.now().millisecondsSinceEpoch;
    transferred = 0;
  }

  // 带参构造
  TransferTask.create(String targetIp, String fileName, int totalSize, String type) {
    // 调用基础初始化
    taskId = Uuid().v4();
    createTime = DateTime.now().millisecondsSinceEpoch;
    transferred = 0;

    this.targetIp = targetIp;
    this.fileName = fileName;
    this.totalSize = totalSize;
    this.type = type;
  }

  // Getters & Setters
  String getTaskId() => taskId;
  void setTaskId(String taskId) => this.taskId = taskId;

  String? getTargetIp() => targetIp;
  void setTargetIp(String? targetIp) => this.targetIp = targetIp;

  String? getFileName() => fileName;
  void setFileName(String? fileName) => this.fileName = fileName;

  int getTotalSize() => totalSize;
  void setTotalSize(int totalSize) => this.totalSize = totalSize;

  int getTransferred() => transferred;
  void setTransferred(int transferred) => this.transferred = transferred;

  String? getStatus() => status;
  void setStatus(String? status) => this.status = status;

  String? getType() => type;
  void setType(String? type) => this.type = type;

  int getCreateTime() => createTime;

  // 获取进度百分比
  int getProgress() {
    if (totalSize == 0) return 0;
    return (transferred * 100) ~/ totalSize;
  }
}